src/cart/crt_hg.c

```diff
@@ -1309,7 +1309,17 @@ crt_hg_req_send_cb(const struct hg_cb_info *hg_cbinfo)
 		break;
 	default:
 		rpc_priv->crp_state = RPC_STATE_COMPLETED;
-		rc = crt_hgret_2_der(hg_cbinfo->ret);
+
+		if (crt_is_service()) {
```

Comment:
[Question] Why do we differentiate "service" and "non-service" cases? I'm curious because part of libdaos code is also used in a daos_engine, where `crt_is_service` returns true. (Is that right, @wangdi1?)

Reply:
the idea here is that client will just get all errors squashed as 'couldnt send', as a client shouldnt really care about why the destination was not reachable, while server might need an exact error

Reply:
I see. Just wanted to remind that the `daos` "client" code executes both in "service" and "non-service" cases.

Reply:
The engines essentially do the old client behavior (receive and evaluate detailed results from the send) whereas the the clients now do a simplified handling (retrying on all failures). This makes sense conceptually, and it allows all of the daos_test suite to pass (without the crt_is_service check, DTX and rebuild tests were failing).